### PR TITLE
Add readme and snippets to the React Native getting started

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do?
This PR add the snippets and the Readme of the react native getting started.

### How should this be manually tested?
Check Netlify or the _doc/7/getting-started/react-native/index.md_

### Other changes
/

### Boyscout
/

### CODE REVIEW

File: doc/7/getting-started/react-native/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+code: false
+title: React Native
+description: Getting started with Kuzzle and React Native using Expo
+order: 400
+---
+
+
+# Getting Started with Kuzzle and React Native
+
+This section deals with **Kuzzle** (+ **Javascript SDK**) and **React Native**. We will create **documents** in Kuzzle and subscribe to [document notifications](/sdk/js/7/essentials/realtime-notifications/#document-messages) to develop a realtime chat.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]: Maybe a link to the SDK doc ?

File: doc/7/getting-started/react-native/index.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+- **Expo CLI** ([install here](https://docs.expo.io/versions/v36.0.0/get-started/installation/))
+- **Running Kuzzle Stack** ([instructions here](/core/2/guides/getting-started/running-kuzzle/))
+
+## Prepare your environment
+
+Create your React Native app with Expo CLI.
+```bash
+expo init getting-started-kuzzle
+```
+You can use yarn to install dependancies if you want.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
You can use yarn to install dependencies if you want.
```

File: doc/7/getting-started/react-native/index.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+Then choose a blank project, type 'Kuzzle' and type y.
+
+Install Kuzzle's Javascript SDK: 
+```bash
+cd getting-started-kuzzle
+npm install kuzzle-sdk
+```
+
+Now, you can run your app and access if by different way:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Now, you can run your app and access it by different ways:
```

File: doc/7/getting-started/react-native/index.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  expo start --web
+  ```
+
+## Get the username
+First of all, we need to know the user's name.
+Let's start by creating the following file _screens/Login/Login.js_ and add some imports.
+
+<<< ./snippets/Login.js.snippet:1[js]
+
+Then we must to export our `Login` class with the render function:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Then we must export our `Login` class with the render function:
```

File: doc/7/getting-started/react-native/index.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  expo start
+  ```
+  - To test it in your browser, you can run the following command:
+  ```bash
+  expo start --web
+  ```
+
+## Get the username
+First of all, we need to know the user's name.
+Let's start by creating the following file _screens/Login/Login.js_ and add some imports.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Let's start creating the following file _screens/Login/Login.js_ and add some imports.
```

File: doc/7/getting-started/react-native/index.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+<<< ./snippets/Login.js.snippet:1[js]
+
+Then we must to export our `Login` class with the render function:
+
+<<< ./snippets/Login.js.snippet:2[js]
+
+As you can see, this component has no state. It'll send the typed username
+to his parent composent by a `onSubmitName` method passed as props.
+
+To finish, we can add some style properties to let our login page looks better.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
To finish, we can add some style properties to let our login page looks prettier.
```

File: doc/7/getting-started/react-native/index.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+Let's start by creating the following file _screens/Login/Login.js_ and add some imports.
+
+<<< ./snippets/Login.js.snippet:1[js]
+
+Then we must to export our `Login` class with the render function:
+
+<<< ./snippets/Login.js.snippet:2[js]
+
+As you can see, this component has no state. It'll send the typed username
+to his parent composent by a `onSubmitName` method passed as props.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
to his parent component by a `onSubmitName` method passed as props.
```

File: doc/7/getting-started/react-native/index.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+This parent component of _Login.js_ will be _App.js_ so let's work on it. 
+
+You can delete all the code from the _App.js_ file, we will rewrite it.
+
+First, add the following imports:
+
+<<< ./snippets/App.js.snippet:1[js]
+
+Then we need to do three changes:
+  - In first, we must export our *App* class:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
  - First, we must export our *App* class:
```

File: doc/7/getting-started/react-native/index.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    <<< ./snippets/App.js.snippet:2[js]
+  - In second, we need to create our `onSubmitName` function:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
  - Then we need to create our `onSubmitName` function:
```

File: doc/7/getting-started/react-native/index.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    <<< ./snippets/App.js.snippet:3[js]
+  - To finish, we will create our render function:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
  - Finally we will create our render function:
```

File: doc/7/getting-started/react-native/index.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    <<< ./snippets/App.js.snippet:4[js]
+
+You can see that we will have to create a new component, *ChatClient*.
+That new component will contain the link to our kuzzle backend so we need first declare our kuzzle service.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
That new component will contain the link to our Kuzzle backend so we need first declare our Kuzzle service.
```

File: doc/7/getting-started/react-native/index.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+
+## Instantiating Kuzzle SDK and Adding logical part
+
+We have to connect the server so that our client can interact with it.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
We have to connect the server so our client can interact with it.
```

File: doc/7/getting-started/react-native/index.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+
+To do this, we have to create a _/services/kuzzle.js_ file to put our kuzzle instance declaration:
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
To do this, we have to create a _/services/kuzzle.js_ file to put our Kuzzle instance declaration:
```

File: doc/7/getting-started/react-native/index.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+
+Note that if you are running your application on your smartphone, you must be connected on the same network and you'll need to change the `localhost` in the previous snippet by your computer ip address.
+
+We can now create a _screens/Chat/ChatClient.js_ file to use our kuzzle service.
+
+Like in our previous components, we'll first need to add our imports:
+
+<<< ./snippets/ChatClient.js.snippet:1[js]
+
+Then we will create our state that will contains our messages and call the function that will initialize our kuzzle by [establish the connection](/sdk/js/7/core-classes/kuzzle/connect/) to kuzzle and create, if they don't [exist](sdk/js/6/controllers/index/exists/), the [index](sdk/js/6/controllers/index/create/) and [collection](sdk/js/6/controllers/collection/create/) of our chat.
```
Comment (Aschen) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Then we will create our state that will contains our messages and call the function that will initialize our kuzzle by [establish the connection](/sdk/js/7/core-classes/kuzzle/connect/) to kuzzle and create, if they don't [exist](sdk/js/7/controllers/index/exists/), the [index](sdk/js/6/controllers/index/create/) and [collection](sdk/js/7/controllers/collection/create/) of our chat. 
```

File: doc/7/getting-started/react-native/index.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+As you can see we don't push the new message in our array on message creation.
+
+Indeed, we will receive notifications from Kuzzle each time we modify our message collection (even if it is a message creation on our part) that we will use to add the messages in our array.
+
+Since this component is used to do the connection to Kuzzle, it will render another one that will display the messages _screens/Chat/ChatView.js_. 
+To finish, we just need to add the `render()` function: 
+
+<<< ./snippets/ChatClient.js.snippet:9[js]
+
+As you can see, we are passing as props the fetched messages, the client username and the handleSendMessage function.
```
Comment (Aschen) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
As you can see, we are passing as props the fetched messages, the client username and the `handleSendMessage` function.
```

File: doc/7/getting-started/react-native/index.md [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+Now we need to create ou `ChatView` class and bind a `handleSendMessage` property to an `onSendMessage` function that will call the props parent function. 
+
+<<< ./snippets/ChatView.js.snippet:2[js]
+
+There is no logic in this component so let's work on the render function:
+
+<<< ./snippets/ChatView.js.snippet:5[js]
+
+In the previous snippet, you can see that we are using a Flatlist to display the messages passed as props from the `ChatClient` component.
```
Comment (Aschen) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
In the previous snippet, you can see that we are using a `Flatlist` to display the messages passed as props from the `ChatClient` component.
```

File: doc/7/getting-started/react-native/index.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+The first view part in the Flatlist to the previous optionnal part. It is used to display the date only once for each message sended at the same date.
```
Comment (Aschen) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
The first view part in the `Flatlist` to the previous optional part. It is used to display the date only once for each message sent at the same date.
```

File: doc/7/getting-started/react-native/index.md [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+
+Then just add the TextInput part with a call to our `handleSendMessage` property:
```
Comment (Aschen) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Then just add the `TextInput` part with a call to our `handleSendMessage` property:
```

File: doc/7/getting-started/react-native/index.md [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+<<< ./snippets/ChatClient.js.snippet:6[js]
+
+The function `fetchMessage()` will [search](/sdk/js/7/controllers/document/search/) for the first hundred newest messages and store them in our state, before subscribing to changes in the `messages` collection.
+
+Then, create the `subscribeMessages()` function.
+It will call the Kuzzle's realtime controller to allow us to [receive notifications](/sdk/js/7/controllers/realtime/subscribe/) on message creations:
+
+<<< ./snippets/ChatClient.js.snippet:7[js]
+
+You can see in the two function we just wrote a call to a `getMessage()` function when we are adding a new message to the state. This function will format our kuzzle's notification/response.
```
Comment (Yoann-Abbes) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
You can see in the two function we just wrote a call to a `getMessage()` function when we are adding a new message to the state. This function will format our Kuzzle's notification/response.
```
You forgot to uppercase Kuzzle in several places

File: doc/7/getting-started/react-native/index.md [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+
+<<< ./snippets/ChatClient.js.snippet:5[js]
+
+Note: The first part of this function is optionnal and can be replace by a set of the `display` variable to false.
```
Comment (xbill82) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
Note: The first part of this function is optional and can be replaced by a set of the `display` variable to false. 
```

File: doc/7/getting-started/react-native/index.md [hunk 21]
```diff
@@ -0,0 +1,187 @@
...
+
+<<< ./snippets/ChatClient.js.snippet:7[js]
+
+You can see in the two function we just wrote a call to a `getMessage()` function when we are adding a new message to the state. This function will format our kuzzle's notification/response.
+
+<<< ./snippets/ChatClient.js.snippet:5[js]
+
+Note: The first part of this function is optionnal and can be replace by a set of the `display` variable to false. 
+
+If you decided to add the optionnal part, you'll need to implement the `displayDate()` function:
```
Comment (xbill82) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
If you decided to add the optional part, you'll need to implement the `displayDate()` function:
```

File: doc/7/getting-started/react-native/index.md [hunk 22]
```diff
@@ -0,0 +1,187 @@
...
+
+<<< ./snippets/ChatClient.js.snippet:4[js]
+
+In the constructor of this component, we setted a `handleSendMessage` property to a binded `onSendMessage` function. Let's implement that function:
```
Comment (xbill82) [inline doc/7/getting-started/react-native/index.md]:
```suggestion
In the constructor of this component, we set a `handleSendMessage` property to a bound `onSendMessage` function. Let's implement that function:
```

**Pull request merged**